Sources/Controllers/TargetMenu/Favorites/OAFavoriteImportViewController.mm

```diff
@@ -10,331 +10,302 @@
+#pragma mark - UITableViewDataSource
 
-- (IBAction) importClicked:(id)sender
+- (UITableViewCell *)getRow:(NSIndexPath *)indexPath
 {
-    if (_favoritesCollection) {
-        // IOS-214
-        if (![self isFavoritesValid])
-            return;
-        
-        _app.favoritesCollection->mergeFrom(_favoritesCollection);
-        [OAFavoritesHelper loadFavorites];
-        // Re-apply parking
-        OAParkingPositionPlugin *plugin = (OAParkingPositionPlugin *)[OAPlugin getPlugin:OAParkingPositionPlugin.class];
-        if (plugin)
+    OATableRowData *item = [self.tableData itemForIndexPath:indexPath];
+    if ([item.cellType isEqualToString:[OAPointTableViewCell getCellIdentifier]])
+    {
+        OAPointTableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:[OAPointTableViewCell getCellIdentifier]];
+        if (!cell)
         {
-            for (OAFavoriteItem *item in [OAFavoritesHelper getFavoriteItems])
-            {
-                if (item.specialPointType == OASpecialPointType.PARKING)
-                {
-                    NSDate *timestamp = [item getTimestamp];
-                    NSDate *pickupTime = [item getPickupTime];
-                    BOOL isTimeRestricted = pickupTime != nil && [pickupTime timeIntervalSince1970] > 0;
-                    [plugin setParkingType:isTimeRestricted];
-                    [plugin setParkingTime:isTimeRestricted ? pickupTime.timeIntervalSince1970 * 1000 : 0];
-                    if (timestamp)
-                        [plugin setParkingStartTime:timestamp.timeIntervalSince1970 * 1000];
-                    [plugin setParkingPosition:item.getLatitude longitude:item.getLongitude];
-                    [plugin addOrRemoveParkingEvent:item.getCalendarEvent];
-                    if (item.getCalendarEvent)
-                        [OAFavoritesHelper addParkingReminderToCalendar];
-                    else
-                        [OAFavoritesHelper removeParkingReminderFromCalendar];
-                    break;
-                }
-            }
+            NSArray *nib = [[NSBundle mainBundle] loadNibNamed:[OAPointTableViewCell getCellIdentifier] owner:self options:nil];
+            cell = nib[0];
+            cell.selectionStyle = UITableViewCellSelectionStyleNone;
         }
-        [self.ignoredNames removeAllObjects];
-        self.conflictedName = @"";
+        if (cell)
+        {
+            OAWptPt *wptPt = [item objForKey:@"wptPt"];
+
+            cell.titleView.text = wptPt.name;
+            cell.rightArrow.image = nil;
+            cell.directionImageView.image = nil;
+            cell.distanceView.hidden = YES;
+
+            CGRect titleFrame = CGRectMake(cell.titleView.frame.origin.x, 15.0, cell.titleView.frame.size.width + 20.0, cell.titleView.frame.size.height);
+            cell.titleView.frame = titleFrame;
 
-        [OAUtilities denyAccessToFile:_url.path removeFromInbox:YES];
-        [OsmAndApp.instance saveFavoritesToPermanentStorage];
```

Comment:
It is not clear why saveFavoritesToPermanentStorage used in other places and should it be replaced with another method or not on this iteration